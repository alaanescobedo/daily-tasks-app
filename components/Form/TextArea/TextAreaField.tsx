import { Stack } from '@components/Stack'
import { forwardRef, useState } from 'react'
import { TextareaBase } from './BaseTextarea/TextareaBase'
import { InputError } from '../Input/InputError/InputError'
import InputField from '../Input/InputField/InputField'
import { InputHint } from '../Input/InputHint/InputHint'
import { InputLabel } from '../Input/InputLabel/InputLabel'

export const TextAreaField = forwardRef(({
  id,
  name,
  error,
  hint,
  label,
  required,
  maxLength = 120,
  resize,
  gap,
  placeholder = 'Write here...',
  onChange = () => { },
  hidenErrorField = false,
  keepHint = false,
  value,
  size,
  ...props
}: any, ref): JSX.Element => {
  const [textareaValue, setTextareaValue] = useState(value ?? '')

  const handleChange = (e: any): void => {
    setTextareaValue(e.target.value)
    onChange(e)
  }

  const fieldProps = {
    id,
    name,
    error,
    hint,
    label,
    required,
    size
  }
  const stackProps = {
    gap
  }

  const baseProps = {
    onChange,
    placeholder,
    maxLength,
    resize,
    value: textareaValue,
    size,
    ...props
  }

  return (
    <InputField {...fieldProps}>
      <Stack {...stackProps}>
        <InputLabel />
        <TextareaBase {...baseProps} onChange={handleChange} ref={ref} />
        <InputHint keepHint={keepHint} />
        {hidenErrorField === false && <InputError />}
      </Stack>
    </InputField>
  )
}
)
