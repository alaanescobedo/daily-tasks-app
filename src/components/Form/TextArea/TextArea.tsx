import { ChangeEvent } from 'react'
import styles from '../shared/styles.module.css'

interface Props {
  value: string
  id: string
  name: string
  placeholder: string
  handleChange: (e: ChangeEvent<HTMLTextAreaElement>) => void
  maxLength?: number
}
export const TextArea = ({ handleChange, value, name, id, maxLength = 70, placeholder = 'Write here...' }: Props): JSX.Element => {
  return (
    <textarea
      onChange={handleChange}
      className={styles.input_task}
      placeholder={placeholder}
      maxLength={maxLength}
      value={value}
      name={name}
      id={id}
    />
  )
}
