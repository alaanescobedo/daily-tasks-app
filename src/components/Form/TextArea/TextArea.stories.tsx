import { ComponentMeta, Story } from '@storybook/react'

import { TextArea, TextAreaProps } from './TextArea'

const InputTextAreaStory: ComponentMeta<typeof TextArea> = {
  title: 'Inputs/Textarea',
  component: TextArea
}
export default InputTextAreaStory

const Template: Story<TextAreaProps> = (args) => <TextArea {...args} />

export const Primary = Template.bind({})
const primaryArgs: TextAreaProps = {
  id: 'textarea',
  handleChange: () => { },
  value: '',
  placeholder: 'Title',
  maxLength: 70
}
Primary.args = primaryArgs
