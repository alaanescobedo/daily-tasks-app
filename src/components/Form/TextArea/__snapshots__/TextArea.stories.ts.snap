// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Form/Textarea Primary 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "--background": "2, 40, 60",
        "--highlight": "255,  202,  16",
        "--opacity-1": 0.1,
        "--opacity-7": 0.7,
        "--primary": "206, 180, 249",
        "--secondary": "189, 231, 255",
        "--tertiary": "176, 239, 165",
        "--text": "245, 245, 245",
        "0.1": "--opacity-1",
        "0.7": "--opacity-7",
      }
    }
  >
     
    <textarea
      className="input_task"
      id="textarea"
      maxLength={70}
      name="textarea"
      onChange={[Function]}
      placeholder="Title"
      value=""
    />
     
  </div>,
  " ",
]
`;
